src/index.ts

```diff
@@ -139,6 +141,27 @@ class MetricsGatherer {
 		};
 	}
 
+	aggregateRequestHandler(authTest? : AuthTestFunc, callback? : Function) : express.Handler {
+		const aggregatorRegistry = new prometheus.AggregatorRegistry();
+		return (req, res) => {
+			if (authTest && !authTest(req)) {
+				res.status(403).send();
+			}
+			aggregatorRegistry.clusterMetrics()
+				.then((metrics: string) => {
+					res.set('Content-Type', aggregatorRegistry.contentType);
+					res.send(metrics);
+					if (callback) {
```

Comment:
What is this callback for?  Also you'd probably want to have it as `.asCallback(callback)` if you want to convert the promise interface to a callback interface (assuming bluebird)

Reply:
Originally, callbacks on successful metrics scrapes were an option because of an existing issue with how the underlying `prom-client` implements percentiles (they are "sticky" and don't respect the `maxAgeSeconds` parameter, so you have to reset them manually after each scrape, until the issue is fixed). But, you know, that's not something good to adapt to, so in fact I'll take the callback option out altogether.